### ISSUE TITLE
armcc gives warning on floating point operation

### ISSUE THREAD

**Issue by emidttun** [ThrowTheSwitch/Unity#110]
I see this has been an issue earlier, but for armcc I still get warnings:
```
"\path\to\unity.c", line 659: Warning:  #222-D: floating-point operation result is out of range
              is_trait = ((1.0f / f_zero) == actual) ? 1 : 0;
                                ^
```
In addition to L659, there is one for L663, and two for L675.

Not really sure what would be the right fix for it.

**Comment by emidttun** [comment]
Additional detail: This warning occurs only when using the `--strict` option.

**Comment by mvandervoord** [comment]
Yeah, I'm open to ideas. There are a couple platforms that have defines for these things already, but that is unfortunately rare. :(

**Comment by emidttun** [comment]
My current favourite is to use a macro from ```math.h```
```
is_trait = isinf(actual) ? 1 : 0;
```
The main drawback might be portability, and since this is a C99 feature it will not work for C90. I tried it for armcc and MinGW, and it worked in both cases.

Otherwise it should be possible to define infinity in hex from the IEEE 754 definition.

**Comment by mvandervoord** [comment]
This has finally been fixed in commit c6dc96f. I'm using the c99 macros when they exist and have added a check for UNITY_EXCLUDE_MATH_H to revert to the old method where it doesn't exist.
